packages/graphql-cli-validate-queries/index.js

```diff
@@ -0,0 +1,119 @@
+        default: ['apollo'],
+        describe: 'GraphQL clients in use in this project.',
+        type: 'array'
+    },
+    filesGlob: {
+        alias: 'f',
+        describe: 'The glob used to target files for validation.',
+        type: 'string'
+    },
+    project: {
+        alias: 'p',
+        describe: 'Project name as specified in graphql config.',
+        type: 'string'
+    }
+};
+
+/**
+ * Validate GraphQL queries against a schema.
+ */
+async function validateQueries(context, argv) {
+    try {
+        console.log('Validating project queries...');
+
+        const { clients, filesGlob, project } = argv;
+
+        // Ensure the schema exists.
+        context.spinner.start('Locating schema...');
+        const projectConfiguration = await context.getProjectConfig();
+        const { schemaPath } = projectConfiguration.config;
+        const schemaExists = fs.existsSync(schemaPath);
+        if (!schemaExists) {
+            throw new Error(
+                `Could not find a schema at ${schemaPath}.\n Run \`graphql get-schema --project ${project}\` to download the schema before running ${PLUGIN_COMMAND}.`
+            );
+        }
+        context.spinner.succeed();
+
+        // Validate our queries against that schema.
+        context.spinner.start('Finding queries in files...');
+        const validator = getValidator({ clients, project });
+        const queryFiles = validator.resolveFileGlobPatterns([filesGlob]);
+        context.spinner.succeed();
+
+        context.spinner.start(
+            "Validating project's queries against the schema..."
+        );
+        const report = validator.executeOnFiles(queryFiles);
+        context.spinner.succeed();
+
+        console.log(`Checked ${report.results.length} files.`);
+
+        // Report the results.
+        if (report.errorCount === 0) {
+            console.log('All queries are valid.');
+        } else {
+            console.warn('Found some invalid queries:');
+
+            const formatter = validator.getFormatter();
+            console.log(formatter(report.results));
+        }
+
+        process.exit(0);
+    } catch (e) {
+        const message = e.message ? e.message : e;
+        context.spinner.fail(`An error occurred:\n${message}`);
+        process.exit(1);
+    }
+}
+
+/**
+ * Supplies an object containing the arguments that this plugin supports.
+ */
+function getSupportedArguments(args) {
+    return args.options(PLUGIN_SUPPORTED_ARGUMENTS);
+}
+
+function getValidator({ clients, project }) {
+    const clientRules = clients.map(clientName => ({
+        env: clientName,
+        projectName: project
+    }));
+
+    const ruleDefinition = ['error', ...clientRules];
+
+    const linterConfiguration = {
+        parser: 'babel-eslint',
+        plugins: ['graphql'],
+        rules: {
+            'graphql/template-strings': ruleDefinition,
+            'graphql/no-deprecated-fields': ruleDefinition
```

Comment:
There are a few other rules that we could turn on or off ... should we allow the user to configure which rules are included? We could add them to the list of `supportedArguments`.

Perhaps this can be punted, as its not essential at the moment.

Reply:
Let's punt this, adding a TODO that the supported rules should be configurable on the command line or through .graphqlconfig extensions at some point.

Reply:
Sounds good, I created #1017 to track.